src/textual/_text_area_theme.py

```diff
@@ -100,17 +100,16 @@ def apply_css(self, text_area: TextArea) -> None:
                 self.gutter_style = self.base_style.copy()
 
         background_color = Color.from_rich_color(self.base_style.bgcolor)
-        if self.cursor_style is None:
-            # If the theme doesn't contain a cursor style, fallback to component styles.
-            cursor_style = get_style("text-area--cursor")
-            if cursor_style:
-                self.cursor_style = cursor_style
-            else:
-                # There's no component style either, fallback to a default.
-                self.cursor_style = Style(
-                    color=background_color.rich_color,
-                    bgcolor=background_color.inverse.rich_color,
-                )
+        # If the theme doesn't contain a cursor style, fallback to component styles.
+        cursor_style = get_style("text-area--cursor")
+        if cursor_style:
+            self.cursor_style = cursor_style
+        else:
+            # There's no component style either, fallback to a default.
+            self.cursor_style = Style(
```

Comment:
Style.from_colors is an alternative constructor that's a little faster.

Reply:
The type signature for that doesn't align - it only supports `Color | None`, not `Color | str | None` like `__init__` does.

Reply:
Yes, but you are using Rich `Color` instances here aren't you? Should be a drop-in replacement.

Granted this doesn't look like time critical code.

Reply:
Ah I see, yes in this particular instance I could swap it. However, it only runs once.